src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -1296,10 +1296,12 @@ public PropertySpec(string propertyName, PropertySpec? next)
                     Type? objType = obj?.GetType();
                     if (fetch == null || fetch.Type != objType)
                     {
-                        _fetchForExpectedType = fetch = PropertyFetch.FetcherForProperty(
-                            objType, _propertyName);
+                        _fetchForExpectedType = fetch = PropertyFetch.FetcherForProperty(objType, _propertyName);
                     }
-                    return fetch!.Fetch(obj);
+                    object? ret = null;
+                    // Avoid the exception which can be thrown during accessing the object properties.
```

Comment:
What exception?  Or does this just mean any property might throw any exception and we want to just ignore any properties that fail?